File: backend/spring/clean-architecture/full/application/src/main/java/apptemplate/application/dto/file/UploadedFileDto.java
package apptemplate.application.dto.file;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UploadedFileDto {
    private Long id;
    private String fileName;
    private String originalFileName;
    private String contentType;
    private Long fileSize;
    private String description;
    private String category;
    private boolean isPublic;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Long createdBy;
    private String downloadUrl;
}

File: backend/dotnet/clean-architecture/minimal/src/Core/App.Template.Application/DTOs/UploadedFileDto.cs
namespace AppTemplate.Application.DTOs;

public class UploadedFileDto
{
    public long Id { get; set; }
    public string FileName { get; set; } = string.Empty;
    public string OriginalFileName { get; set; } = string.Empty;
    public string ContentType { get; set; } = string.Empty;
    public long FileSize { get; set; }
    public string? Description { get; set; }
    public string? Category { get; set; }
    public bool IsPublic { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string? CreatedBy { get; set; }
    public string DownloadUrl { get; set; } = string.Empty;
}

public class UpdateFileRequest
{
    public string? Description { get; set; }
    public string? Category { get; set; }
    public bool? IsPublic { get; set; }
}
